keel-ec2-plugin/src/test/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandlerTests.kt

```diff
@@ -395,6 +397,59 @@ internal class ClusterHandlerTests : JUnit5Minutests {
       }
     }
 
+    context("the cluster has active server groups with missing app version tag in one region") {
+      before {
+        coEvery { cloudDriverService.activeServerGroup("us-east-1") } returns activeServerGroupResponseEast
+        coEvery { cloudDriverService.activeServerGroup("us-west-2") } answers { activeServerGroupResponseWest.withMissingAppVersion() }
+      }
+
+      test("app version is null in the region with missing tag") {
+        val current = runBlocking {
+          current(resource)
+        }
+        expectThat(current).containsKey("us-west-2")
+        expectThat(current["us-west-2"]!!.launchConfiguration.appVersion).isNull()
+      }
+
+      test("no exception is thrown") {
+        expectCatching {
+          runBlocking {
+            current(resource)
+          }
+        }.succeeded()
+      }
+
+      test("no event is fired indicating an app version is deployed") {
+        runBlocking {
+          current(resource)
+        }
+        verify(exactly = 0) { publisher.publishEvent(ofType<ArtifactVersionDeployed>()) }
+      }
+
+      test("applying the diff creates a server group") {
+        val modified = setOf(
+          serverGroupEast.copy(name = activeServerGroupResponseEast.name),
+          serverGroupWest.copy(name = activeServerGroupResponseWest.name).withMissingAppVersion()
+        )
+        val diff = ResourceDiff(
+          serverGroups.byRegion(),
+          modified.byRegion()
+        )
+        runBlocking {
+          upsert(resource, diff)
+        }
+
+        val slot = slot<OrchestrationRequest>()
+        coVerify { orcaService.orchestrate("keel@spinnaker", capture(slot)) }
+
+        expectThat(slot.captured.job.first()) {
+          get("type").isEqualTo("createServerGroup")
+          // TODO (lpollo): since we rely on the tag when looking up server groups, should we add the tag when creating them?
```

Comment:
@robfletcher Do you have an opinion on this?

Reply:
@robfletcher FWIW, I checked clouddriver and can see where the tag is being used/parsed, but not created/added, so this remains a mystery to me...

Reply:
Yeah, I'm honestly not 100% sure. I _believe_ it's in CloudDriver somewhere.

Reply:
OK. Removing that `TODO` then.